### PR TITLE
Add configurable random number generator

### PR DESCRIPTION
**Pull request by boulter**
Useful for testing when you want to compare against known values.

@hs-lsong @pfarrel @jmagnarelli-hs

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java [hunk 1]
```diff
@@ -75,20 +74,20 @@ public Object filter(Object object, JinjavaInterpreter interpreter, String... ar
...
+      int index = interpreter.getRandom().nextInt(size);
       while (index-- > 0) {
         it.next();
       }
       return it.next();
     }
     // number
     if (object instanceof Number) {
-      return ThreadLocalRandom.current().nextLong(((Number) object).longValue());
+      return interpreter.getRandom().nextInt(((Number) object).intValue());
```
Comment (hs-lsong) [inline src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java]: Why changing long to int here?

File: src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java [hunk 2]
```diff
@@ -75,20 +74,20 @@ public Object filter(Object object, JinjavaInterpreter interpreter, String... ar
...
     }
     // string
     if (object instanceof String) {
       try {
-        return ThreadLocalRandom.current().nextLong(new BigDecimal((String) object).longValue());
+        return interpreter.getRandom().nextInt(new BigDecimal((String) object).intValue());
```
Comment (hs-lsong) [inline src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java]: Same here.

File: src/test/java/com/hubspot/jinjava/lib/filter/ShuffleFilterTest.java [hunk 3]
```diff
@@ -2,27 +2,31 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.failBecauseExceptionWasNotThrown;
+import static org.mockito.Mockito.*;
```
Comment (hs-lsong) [inline src/test/java/com/hubspot/jinjava/lib/filter/ShuffleFilterTest.java]: import static org.mockito.Mockito.when should be enough.

File: src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java (hunk 1 above)
Comment (boulter) [inline src/main/java/com/hubspot/jinjava/lib/filter/RandomFilter.java]: There's no `nextLong()` that takes a bound in `java.util.Random`. I figured random ints are probably big enough.

File: src/test/java/com/hubspot/jinjava/lib/filter/ShuffleFilterTest.java (hunk 3 above)
Comment (boulter) [inline src/test/java/com/hubspot/jinjava/lib/filter/ShuffleFilterTest.java]: Our style is to always collapse imports for Mockito and a few other deps.

**Pull request merged**